src/renderer/sw_engine/tvgSwFill.cpp

```diff
@@ -201,7 +202,17 @@ bool _prepareRadial(SwFill* fill, const RadialGradient* radial, const Matrix* tr
         fill->radial.fy = cy + r * (fy - cy) / dist;
         fill->radial.dx = cx - fill->radial.fx;
         fill->radial.dy = cy - fill->radial.fy;
+#if TVG_FLOAT_CONTRACT_AVAILABLE
         fill->radial.a = fill->radial.dr * fill->radial.dr - fill->radial.dx * fill->radial.dx - fill->radial.dy * fill->radial.dy;
+#else
+        // Prevent loss of precision on Apple Silicon when dr=dy and dx=0 due to FMA
+        // https://github.com/thorvg/thorvg/issues/2014
```

Comment:
one more, could you please remove this issue history? we can track the history by git blame with commit msgs.

Reply:
Thanks for the insights, I've integrated you requests and squashed the history.